OpenLaTeXStudio/EditorModule/src/latexstudio/editor/LaTeXSettingsPanel.java

```diff
@@ -198,6 +193,11 @@ public void run() {
 
     }
     
+    @SettingListener( setting = ApplicationSettings.Setting.AUTOCOMPLETE_ENABLED )

```

Comment:
Not unless the ApplicationSettings.setSettingValue is actually called from the SettingListener annotated method, I think (which here is in store()). And it would seem obvious to me that such a thing is a very bad idea.
I still dislike the disparity between the annotation and the setSettingValue method, though.